src/test/java/com/iexec/worker/replicate/ReplicateDemandServiceTests.java

```diff
@@ -66,28 +68,29 @@ void init() {
         MockitoAnnotations.openMocks(this);
     }
 
+    // region triggerAskForReplicate()
     @Test
-    void shouldRunAskForReplicateAsynchronouslyWhenTriggeredOneTime() throws InterruptedException {
+    void shouldRunAskForReplicateAsynchronouslyWhenTriggeredOneTime() {
         ThreadNameWrapper threadNameWrapper = new ThreadNameWrapper();
         String mainThreadName = Thread.currentThread().getName();
         doAnswer(invocation -> TestUtils.saveThreadNameThenCallRealMethod(threadNameWrapper, invocation))
                 .when(replicateDemandService).askForReplicate();
-        
-        replicateDemandService.triggerAskForReplicate();
-        TimeUnit.MILLISECONDS.sleep(10);
+
+        waitForTriggerAskForReplicateEnd(replicateDemandService.triggerAskForReplicate());
```

Comment:
```suggestion
        replicateDemandService.triggerAskForReplicate().join(1000);
```

Reply:
There's no such `join(long timeout)` method, but I found an alternate `completeOnTimeout` that should work! :)